src/db/postgres/lrsql/postgres/record.clj

```diff
@@ -224,4 +231,54 @@
   (-query-platform-frequency [_ tx]
     (query-platform-frequency tx))
   (-query-timeline [_ tx input]
-    (query-timeline tx input)))
+    (query-timeline tx input))
+
+  bp/ReactionBackend
+  (-insert-reaction! [_ tx params]
+    (try
+      (insert-reaction! tx params)
+      (catch PSQLException ex
+        (if (= "23505" (.getSQLState ex))
+          :lrsql.reaction/title-conflict-error
+          (throw ex)))))
+  (-update-reaction! [_ tx params]
+    (try
+      (update-reaction! tx params)
+      (catch PSQLException ex
+        (if (= "23505" (.getSQLState ex))
+          :lrsql.reaction/title-conflict-error
+          (throw ex)))))
+  (-delete-reaction! [_ tx params]
+    (delete-reaction! tx params))
+  (-error-reaction! [_ tx params]
+    (error-reaction! tx params))
+  (-snip-json-extract [_ {:keys [datatype] :as params}]
+    (if (-> tuning :config :enable-jsonb)
+      (snip-jsonb-extract (assoc params :type (datatype pd/type->pg-type)))
+      (snip-json-extract (assoc params :type (datatype pd/type->pg-type)))))
```

Comment:
Ok these line confused me because at first I thought that `datatype` was a function that took `pd/type->pg-type` as a higher order function, so it took me a short while to realize it was a keyword. Reversing the expr order to `(pd/type->pg-type datatype)` should reduce that confusion while being semantically identical.

Reply:
eec03eda1a2de5aaf26b06181d3ce82828331258